### PR TITLE
Remove built-in definition of xml.xsd from xmerl

### PR DESCRIPTION
**Pull request by schlagert**
Removes the built-in definitions of http://www.w3.org/2001/xml.xsd from the `xmerl_xsd` module. `xmerl_xsd` does load the definitions of the above schema into its processing table during initialization. This leads to the inability to process XSD schemas that properly import [xml.xsd](http://www.w3.org/2001/xml.xsd) (as suggested in the schema itself). The processing fails with `not_uniquely_defined_in_schema` errors because the table already contains unusable values associated with valid keys corresponding to types from [xml.xsd](http://www.w3.org/2001/xml.xsd).

The definitions and thus the invalid values are loaded from a binary containing the terms corresponding to the types from [xml.xsd](http://www.w3.org/2001/xml.xsd) in *Erlang external term format*. Since this is not refactoring safe the intended functionality (if any) already broke long ago (prior to [github](https://github.com/) import) because some of the record definitions did change in releases older than `R13B03`.

To sum up, the current code does not do anything useful because the built-in definitions from the binary can't be used due to incompatible records. Thus, this fix is not backwards incompatible. The change also adds two new test cases covering common use-cases when using `xml:` types in custom schemas.

### CODE REVIEW

**Comment by OTP-Maintainer** [comment]
Patch has passed first testings and has been assigned to be reviewed


----------------

I am a script, I am not human 

----------------

**Comment by schlagert** [comment]
Just in case my above description is a bit unclear. With built-in definitions I refer to the binary contained inline in `xmerl_xsd.erl` starting at line 5623. The following is the result when converting the binary into an Erlang term using `binary_to_term/1` as the code does:
```erlang
[{{attribute,
      {space,[],'http://www.w3.org/XML/1998/namespace'}},
  {schema_attribute,
      {space,[],'http://www.w3.org/XML/1998/namespace'},
      [{simpleType,{'_xmerl_no_name_',[space],[]}}],
      false,[],optional,undefined,undefined,undefined}},
 {{schema,"xml.xsd"},
  {schema,unqualified,unqualified,
      'http://www.w3.org/XML/1998/namespace',[],[],[]}},
 {{attribute,
      {base,[],'http://www.w3.org/XML/1998/namespace'}},
  {schema_attribute,
      {base,[],'http://www.w3.org/XML/1998/namespace'},
      [{simpleType,
           {anyURI,[],'http://www.w3.org/2001/XMLSchema'}}],
      false,[],optional,undefined,undefined,undefined}},
 {{attributeGroup,
      {specialAttrs,[],'http://www.w3.org/XML/1998/namespace'}},
  {schema_attribute_group,
      {specialAttrs,[],'http://www.w3.org/XML/1998/namespace'},
      undefined,undefined,
      [{attribute,{base,[],[]}},
       {attribute,{lang,[],[]}},
       {attribute,{space,[],[]}}]}},
 {{simpleType,{'_xmerl_no_name_',[space],[]}},
  {schema_simple_type,
      {'_xmerl_no_name_',[space],[]},
      [space],
      {'NCName',[],'http://www.w3.org/2001/XMLSchema'},
      false,[],
      [{enumeration,["default","preserve"]}],
      atomic,
      [{restriction,
           {{'NCName',[],'http://www.w3.org/2001/XMLSchema'},
            [{enumeration,"default"},{enumeration,"preserve"}]}}]}},
 {{simpleType,{'_xmerl_no_name_',[lang],[]}},
  {schema_simple_type,
      {'_xmerl_no_name_',[lang],[]},
      [lang],
      undefined,false,[],[],atomic,
      [{union,
           [{simpleType,
                {language,[],'http://www.w3.org/2001/XMLSchema'}},
            {simpleType,
                {'_xmerl_no_name_',['_xmerl_no_name_',lang],[]}}]}]}},
 {{attribute,{id,[],'http://www.w3.org/XML/1998/namespace'}},
  {schema_attribute,
      {id,[],'http://www.w3.org/XML/1998/namespace'},
      [{simpleType,{'ID',[],'http://www.w3.org/2001/XMLSchema'}}],
      false,[],optional,undefined,undefined,undefined}},
 {{attribute,
      {lang,[],'http://www.w3.org/XML/1998/namespace'}},
  {schema_attribute,
      {lang,[],'http://www.w3.org/XML/1998/namespace'},
      [{simpleType,{'_xmerl_no_name_',[lang],[]}}],
      false,[],optional,undefined,undefined,undefined}},
 {{simpleType,
      {'_xmerl_no_name_',['_xmerl_no_name_',lang],[]}},
  {schema_simple_type,
      {'_xmerl_no_name_',['_xmerl_no_name_',lang],[]},
      ['_xmerl_no_name_',lang],
      {string,[],'http://www.w3.org/2001/XMLSchema'},
      false,[],
      [{enumeration,[[]]}],
      atomic,
      [{restriction,
           {{string,[],'http://www.w3.org/2001/XMLSchema'},
            [{enumeration,[]}]}}]}}]
```

Now let's have a look at the definition of e.g. the `schema_attribute` record defined in `xmerl_xsd.hrl`:
```erlang
-record(schema_attribute,{
	  name,
	  type,
	  resolved=false,
	  scope,
%%	  required=false,
	  use=optional,     %% optional | required | prohibited
	  default,
	  fixed,
	  form, %% qualified | unqualified
	  id
	 }).
```

Obviously this is not compatible with, e.g.
```erlang
{schema_attribute,
      {lang,[],'http://www.w3.org/XML/1998/namespace'},
      [{simpleType,{'_xmerl_no_name_',[lang],[]}}],
      false,[],optional,undefined,undefined,undefined}}
```

As far as I can see the `schema_attribute` record did not change since `OTP` was imported to GitHub.
